cli/src/cast.rs

```diff
@@ -710,6 +728,51 @@ where
     })
 }
 
+fn det_base_in(value: &str, base_in: Option<String>) -> eyre::Result<u32> {
+    match base_in {
+        Some(base_in) => match base_in.as_str() {
+            "10" | "dec" => Ok(10),
+            "16" | "hex" => Ok(16),
+            _ => Err(eyre::eyre!("could not parse hex base in")),
+        },
+        None => {
+            if value.starts_with("0x") {
+                U256::from_str_radix(value, 16).expect("could not parse hex base in");
+
+                Ok(16)
+            } else {
+                match U256::from_str_radix(value, 10) {
+                    Ok(_) => {
+                        U256::from_str_radix(value, 16)
+                            .expect("could not parse base in, please prepend with 0x if hex.");
+
+                        Ok(10)
+                    }
+                    Err(_) => {
+                        U256::from_str_radix(value, 16).expect("couldn't parse base in from hex");
+                        Ok(16)
+                    }
+                }
+            }
+        }
+    }
+}
+
+fn det_base_out(base_out: &str) -> u32 {
+    match base_out {
+        "10" | "dec" => 10,
+        "16" | "hex" | _ => 16,
+    }
+}
+
+fn format_uint(val: U256, base_out: u32) -> eyre::Result<String> {
+    match base_out {
+        10 => Ok(val.to_string()),
+        16 => Ok(AbiEncode::encode_hex(val)),
```

Comment:
should work, `U256` impls `LowerHex`

```suggestion
        16 => Ok(format!("0x{:x}", val)),
```